CHANGELOG.md

```diff
@@ -20,6 +20,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - _Upgrade (experimental)_: Migrate `@media screen(…)` when running codemods ([#14603](https://github.com/tailwindlabs/tailwindcss/pull/14603))
 - _Upgrade (experimental)_: Inject `@config "…"` when a `tailwind.config.{js,ts,…}` is detected ([#14635](https://github.com/tailwindlabs/tailwindcss/pull/14635))
 - _Upgrade (experimental)_: Migrate `aria-*`, `data-*`, and `supports-*` variants from arbitrary values to bare values ([#14644](https://github.com/tailwindlabs/tailwindcss/pull/14644))
+- _Upgrade (experimental)_: Migrate `theme(…)` calls to `var(…)` or the modern `theme(…)` syntax ([#14664](https://github.com/tailwindlabs/tailwindcss/pull/14664))
```

Comment:
The change log entry is a bit misleading since I would assume all `theme()` functions are migrated when it's only `theme()` function inside candidates.

Might want to clarify this or (even better) follow up with the CSS `theme()` function implementation right away? hopefully should be able to reuse most of the implementation.

Reply:
> The change log entry is a bit misleading since I would assume all `theme()` functions are migrated when it's only `theme()` function inside candidates.

Do you mean that you expect all the `theme()` functions to be gone? Because in case you have `max-[theme(spacing.4)]:flex` then we can't replace the `theme()` with var because the browser doesn't support this. We _could_ sort of resolve the `theme(spacing.4)` value, but then we end up with `max-[1rem]:flex` which isn't "dynamic" anymore if you change your spacing scale.

So in this case, we migrate `max-[theme(spacing.4)]` to the modern theme function `max-[theme(--spacing.4)]`. 

Is this what you mean?

Reply:
Got it!

It's only about the `theme(…)` in classes, _not_ about the `theme(…)` in CSS.